tests/hip/common/common.h

```diff
@@ -116,7 +116,34 @@ hip_test_device_bo {
   T *&get() {
     return _buffer;
   }
+};
+
+// Abstraction of host buffer so we can do automatic buffer dealocation (RAII)
+template<typename T> class
+hip_test_host_bo {
+  T *_buffer;
+
+public:
+  explicit hip_test_host_bo(size_t size, unsigned int flags) : _buffer(nullptr) {
+    test_hip_check(hipHostMalloc((void**)&_buffer, size * sizeof(T), flags));
+  }
+
+  ~hip_test_host_bo() noexcept {
+    test_hip_check(hipHostFree(_buffer));
+  }
```

Comment:
Dtors are automatically noexcept.  The program will crash if `test_hip_check` throws.  If the dtor executes any code that can throw, then it would be better to try/catch and attempt to exit grace fully.
```suggestion
  ~hip_test_host_bo()  {
    test_hip_check(hipHostFree(_buffer));
  }
```